File: app/src/main/java/com/cooper/wheellog/ViewPageAdapter.java
package com.cooper.wheellog;

import android.app.Activity;
import android.view.View;
import android.view.ViewGroup;

import androidx.viewpager.widget.PagerAdapter;

import java.util.ArrayList;

class ViewPageAdapter extends PagerAdapter {

    Activity mActivity;

    private ArrayList<Integer> pages;

    public ViewPageAdapter(Activity activity) {
        mActivity = activity;
        pages = new ArrayList<>();
        reset(true);
    }

    @Override
    public Object instantiateItem(ViewGroup collection, int position) {
        return mActivity.findViewById(pages.get(position));
    }

    public void reset(boolean init) {
        int oldSize = pages.size();
        pages.clear();
        pages.add(R.id.page_one);
        pages.add(R.id.page_two);
        pages.add(R.id.page_three);
        if (!init && oldSize != pages.size()) {
            notifyDataSetChanged();
        }
    }

    public void deletePage(int pageId) {
        if (pages.contains(pageId)) {
            pages.remove((Object) pageId);
            notifyDataSetChanged();
        }
    }

    public void addPage(int pageId) {
        if (!pages.contains(pageId)) {
            pages.add(pageId);
            notifyDataSetChanged();
        }
    }

    @Override
    public int getCount() {
        return pages.size();
    }

    @Override
    public boolean isViewFromObject(View view, Object object) {
        return view == object;
    }
}

File: app/src/main/java/com/cooper/wheellog/ViewPageAdapter.kt
package com.cooper.wheellog

import android.app.Activity
import android.view.View
import android.view.ViewGroup
import androidx.viewpager.widget.PagerAdapter
import com.viewpagerindicator.PageIndicator
import java.util.*

internal class ViewPageAdapter(var mActivity: Activity) : PagerAdapter() {
    private val pages: ArrayList<Int> = ArrayList()
    private var pageIndicator: PageIndicator? = null

    fun setPageIndicator(value: PageIndicator?) {
        pageIndicator = value
    }

    override fun instantiateItem(collection: ViewGroup, position: Int): Any {
        return mActivity.findViewById(pages[position])
    }

    fun hidePage(pageId: Int) {
        if (pages.contains(pageId)) {
            pages.remove(pageId)
            notifyDataSetChanged()
        }
    }

    fun showPage(pageId: Int) {
        showPage(pageId, -1)
    }

    fun showPage(pageId: Int, index: Int) {
        if (!pages.contains(pageId)) {
            if (index == -1) {
                pages.add(pageId)
            } else {
                pages.add(index, pageId)
            }
            notifyDataSetChanged()
        }
    }

    fun isShowed(pageId: Int): Boolean {
        return pages.contains(pageId)
    }

    fun getPageIdByPosition(position: Int): Int {
        return pages[position]
    }

    override fun notifyDataSetChanged() {
        super.notifyDataSetChanged()
        pageIndicator?.notifyDataSetChanged()
        if (pageIndicator is View) {
            (pageIndicator as View).invalidate()
        }
    }

    override fun getCount(): Int {
        return pages.size
    }

    override fun isViewFromObject(view: View, `object`: Any): Boolean {
        return view === `object`
    }
}
